### PR TITLE
move development-related docs to new Development folder

### PR DESCRIPTION
**Pull request by aboedo**
Moves development-related docs into Development folder

### CODE REVIEW

File: RevenueCat.xcodeproj/xcshareddata/xcschemes/BackendIntegrationTests.xcscheme [hunk 1]
```diff
@@ -62,7 +62,7 @@
          </BuildableReference>
       </BuildableProductRunnable>
       <StoreKitConfigurationFileReference
-         identifier = "../../TestingApps/PurchaseTester/RevenueCat_IntegrationPurchaseTesterConfiguration.storekit">
+         identifier = "../../Tests/TestingApps/PurchaseTester/RevenueCat_IntegrationPurchaseTesterConfiguration.storekit">
```
Comment (joshdholtz) [inline RevenueCat.xcodeproj/xcshareddata/xcschemes/BackendIntegrationTests.xcscheme:65]: Was this change a side effect from another PR?

File: RevenueCat.xcodeproj/xcshareddata/xcschemes/BackendIntegrationTests.xcscheme [hunk 2]
```diff
@@ -71,15 +71,6 @@
...
       debugDocumentVersioning = "YES">
-      <MacroExpansion>
-         <BuildableReference
-            BuildableIdentifier = "primary"
-            BlueprintIdentifier = "2DE20B7E26409EB7004C597D"
-            BuildableName = "BackendIntegrationTestsHostApp.app"
-            BlueprintName = "BackendIntegrationTestsHostApp"
-            ReferencedContainer = "container:RevenueCat.xcodeproj">
-         </BuildableReference>
-      </MacroExpansion>
```
Comment (joshdholtz) [inline RevenueCat.xcodeproj/xcshareddata/xcschemes/BackendIntegrationTests.xcscheme:82]: Similar question here. This doesn't seem related to development docs 🤷‍♂️  Seems fine but just wanted to make sure its okay to include in this PR!

**Review by joshdholtz — APPROVED** [review]
Seems good! Just some small questions that can be ignored if they aren't important 🙃

File: RevenueCat.xcodeproj/xcshareddata/xcschemes/BackendIntegrationTests.xcscheme (hunk 1 above)
Comment (aboedo) [reply RevenueCat.xcodeproj/xcshareddata/xcschemes/BackendIntegrationTests.xcscheme:65]: I should have added a comment but I forgot 😞
This was left over from #1414, it was an outdated reference

File: RevenueCat.xcodeproj/xcshareddata/xcschemes/BackendIntegrationTests.xcscheme (hunk 2 above)
Comment (aboedo) [reply RevenueCat.xcodeproj/xcshareddata/xcschemes/BackendIntegrationTests.xcscheme:82]: same here, left over from the other PR, but is safe to merge
